### PR TITLE
Initial data structure

### LINKED ISSUE #1: Data-table
As a dev,
* I would like to see the data table with respective resident data
* Use get/patch methods to display or change resident data in an app.

### PR DESCRIPTION
**Pull request by chadrick-d-dev**
### What’s this PR do?
Creates an express api with knex and postgresql for our Silvernest UI, and gets it to display our seed file data.

### Where should the reviewer start?
Start with the first pull request

### How should this be manually tested?
clone down repo, ensure postgresql and knex are installed on your local machine, run npm install, install postico and postgres.app, create server for Silvernest, run `knex migrate:latest` then run `knex seed:run`, then check out the Silvernest server in postico and look for data

### Any background context you want to provide?
We went on journey to create an express server with PostgreSQL and knex
We were finally able to view our seed file data using Postico

### CODE REVIEW

File: data.json [hunk 1]
```diff
@@ -0,0 +1,192 @@
+[
```
Comment (cameronRomo) [inline data.json:1]: I'm not completely sure if we need this file...

Comment (lo-la-do-li) [inline data.json:1]: I don't think we do either, let's be sure and remember to delete it when we clean up code later. I'll make an issue for it on the project board.

File: db/migrations/20210106133650_create-residents.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
+exports.up = function(knex) {
+  return knex.schema
+  .createTable('residents', function (table) {
+    table.increments('id').primary();
+    table.string('name');
```
Comment (cameronRomo) [inline db/migrations/20210106133650_create-residents.js:5]: `table.string` make sense. More research is needed to understand migrating more complex data types.

File: db/migrations/20210107172948_residents3.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
+exports.up = function (knex) {
```
Comment (lo-la-do-li) [inline db/migrations/20210107172948_residents3.js:1]: Third time's a charm! Once we get the server configured, should we consider adding a table for Appointments and linking to resident's id?

File: knexfile.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
+// Update with your config settings.
```
Comment (lo-la-do-li) [inline knexfile.js:1]: Looking good!

**Review by lo-la-do-li — APPROVED** [review]

File: db/seeds/dev/residents.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        help_with_medications: "yes",
+        other_impairments: "visual impairment",
+        previous_career: "Seamstress",
+        interests: "knitting, cats, card games, unsolved mysteries",
+        pets: "one cat",
+        living_preferences: "",
+        semester: "Spring 2021",
+        additional_notes: "I need help feeding my cat, Mittens. Please don't eat peanuts around me, I am terribly allergic."
+      }
+    ])
```
Comment (cameronRomo) [inline db/seeds/dev/residents.js:24]: Maybe it would be good to look into when/if .catch should be chained here.

Comment (lo-la-do-li) [reply db/seeds/dev/residents.js:24]: Good call - I think maybe Promises were necessary in an earlier version of PostgreSQL, but not in our version. We can definitely look into it though in the docs.

**Review by cameronRomo — APPROVED** [review]
This has been a very interesting experience. Glad we've taken the time to try to learn this!

File: data.json (hunk 1 above)
Comment (chadrick-d-dev) [reply data.json:1]: I'm not either

File: db/migrations/20210107172948_residents3.js (hunk 3 above)
Comment (chadrick-d-dev) [reply db/migrations/20210107172948_residents3.js:1]: I think this wouldn't be too difficult to achieve 🤡

File: knexfile.js (hunk 4 above)
Comment (chadrick-d-dev) [reply knexfile.js:1]: 😎

**Pull request merged**